app/R/data.R

```diff
@@ -115,16 +117,18 @@ createS3DataLoader <- function() {
     # names, sizes, and last modified timestamps). Ignores characteristics of
     # bucket and request, including bucket region, name, content type, request
     # date, request ID, etc.
-    if (nrow(df) == 0 || !identical(s3Contents, newS3Contents)) {
+    if (!(targetVariable %in% names(df_list)) ||
+      nrow(df_list[[targetVariable]]) == 0 ||
+      !identical(s3Contents, newS3Contents)) {
       # Save new data and new bucket connection info to vars in env of
-      # `getRecentDataHelper`. They persist between calls to `getRecentData` a
+      # `createS3DataLoader`. They persist between calls to `getRecentData` a
       # la https://stackoverflow.com/questions/1088639/static-variables-in-r
       s3bucket <<- newS3bucket
-      df <<- getAllData(createS3DataFactory(s3bucket))
+      df_list[[targetVariable]] <<- getAllData(createS3DataFactory(s3bucket), targetVariable)
```

Comment:
question: what is the overhead of `createS3DataFactory`?

will we regret having to create one repeatedly for different `targetVariable`s?

Reply:
Ooo, interesting point. There are definitely some objects here that can be added to the cache; we don' t need to be running `createS3DataFactory` or getting S3 bucket contents for every user, let alone every target variable.